### PR TITLE
Implementation of the server api

### PR DESCRIPTION
**Pull request by User12837985123**
Implementation of the endpoints on the server-side of the api for the walking skeleton.

**What to review:**
The only file to review is:  aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs

**How to run: gitbash**
To run the api, you can simply pull the branch and then run the following command ./build.bat and then the command it tells you to run. 

**How to run: visual studio**
Open the solution file: IO.Swagger.sln. And if it complains about the something being incompatible, just click okay and then later in the solution explorer remove the item and then add it again (right click -> add existing item). Then it should work.

### CODE REVIEW

File: aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        /// <response code="404">no location associated with devive ID</response>
+        [HttpDelete]
+        [Route("/Location/{deviceId}")]
+        [ValidateModelState]
+        [SwaggerOperation("DeleteLocation")]
+        public virtual IActionResult DeleteLocation([FromRoute][Required]int? deviceId)
+        {
+            // TODO: Initiate client elsewhere
+            MongoClient dbClient = new MongoClient("mongodb://admin:mongo801@130.225.57.95:27017");
+            var db = dbClient.GetDatabase("walkingskeleton");
```
Comment (Bjerkk) [inline aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs]: Tænker det her kunne gøres et andet sted i stedet for hvert endpoint, så vi kun skal gøre det en gang. Det gælder selvfølgelig også de andre steder det er skrevet.

Comment (Bjerkk) [reply aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs]: Bare lige når vi skal lave the real deal.

File: aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                {"id", body.Id},
+                {"deviceId", body.DeviceId},
+                {"locationValue", body.LocationValue},
+			};
+
+            // TODO: Check if it was successful.
+            locations.InsertOne(locationDocument);
+
+            // It seems like incorrect input is handled automatically and an error (400) is returned
+            return this.Ok("Done");
```
Comment (Bjerkk) [inline aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs:168]: Så den returnerer også 201 hvis det var gucci?

**Review by Bjerkk — DISMISSED** [review]

File: aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs (hunk 2 above)
Comment (Dbraae) [reply aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs:168]: Yes

File: aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        /// <response code="400">deletion of location went wrong</response>
+        /// <response code="404">no location associated with devive ID</response>
+        [HttpDelete]
+        [Route("/Location/{deviceId}")]
+        [ValidateModelState]
+        [SwaggerOperation("DeleteLocation")]
+        public virtual IActionResult DeleteLocation([FromRoute][Required]int? deviceId)
+        {
+            // TODO: Initiate client elsewhere
+            MongoClient dbClient = new MongoClient("mongodb://admin:mongo801@130.225.57.95:27017");
```
Comment (KVISDAOWNER) [inline aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs]: Make a String const for "mongodb://admin:mongo801@130.225.57.95:27017"

Comment (KVISDAOWNER) [reply aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs]: It is duplicated 4 times :D

File: aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs (hunk 1 above)
Comment (KVISDAOWNER) [reply aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs]: +1 to Bjarke's suggestion. It should be done in the constructor imo.

**Review by KVISDAOWNER — DISMISSED** [review]
It is not possible to run and test the endpoints as the MongoDB is not running on the server, therefor my review is limited. The project compiled okay otherwise. I suggested 2 minor changes to DefaultApi that I think appropriate even for a "WalkingSkeleton". You can do them now before the merge or at some later point, idc.

File: aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs (hunk 3 above)
Comment (User12837985123) [reply aspnet-server-1/src/IO.Swagger/Controllers/DefaultApi.cs]: Thought about it aswell, so I will do it :D

**Review by Bjerkk — APPROVED** [review]

**Review by KVISDAOWNER — APPROVED** [review]

**Pull request merged**